animations/lib/src/misc/curved_animation.dart

```diff
@@ -0,0 +1,146 @@
+// Copyright 2019 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'dart:math' as math;
+
+class CurvedAnimationDemo extends StatefulWidget {
+  static const String routeName = '/misc/curved_animation';
+  @override
+  _CurvedAnimationDemoState createState() => _CurvedAnimationDemoState();
+}
+
+class MyCurve {
+  Curve curve;
+  String curveName;
+  MyCurve({this.curve, this.curveName});
+}
+
+class _CurvedAnimationDemoState extends State<CurvedAnimationDemo>
+    with SingleTickerProviderStateMixin {
+  static const String routeName = '/misc/curved_animation';
+  AnimationController animController;
+  Animation<double> animation;
+  static const _duration = Duration(seconds: 4);
+  TextStyle style = TextStyle(fontSize: 18.0);
+  List<MyCurve> forwardCurves = [
```

Comment:
Out of curiosity, why are there two different lists of curves here? All of them would be valid for use as either the forward or reverse curve as far as the CurvedAnimation widget is concerned.